### DISCUSSION TITLE
Could these effects work in a browser on div's?

### DISCUSSION THREAD

**Discussion by DustinBrett** [Schneegans/Burn-My-Windows#358] [Q&A] [answered]
These effects are amazing and I would love to apply them to my [desktop environment in the browser](https://github.com/DustinBrett/daedalOS). I see some web technologies are being used and I was wondering if there is any guidance on the idea/implementation of applying these effects on the DOM directly to elements such as `<div />`'s. I'd be fine if I had to convert the `div` to a `canvas` or something beforehand, but getting the effects into a browser is what I am less sure how to do atm.

Thanks for any help/suggestions!

**Comment by Schneegans** [comment, marked as answer]
Hi there! Thanks for getting in touch; I have played around with daedalOS before and it's an incredibly awesome project! Using the Burn-My-Windows effects is a cool idea and definitely possible. However, it's not going to be exactly easy, I guess. The rough steps would be:
1. Turn the window div to a texture. This is not straight-forward but somehow possible. Maybe you have already implemented some kind of screen-shot functionality into daedalOS?
2. Hide the window div and render the captured texture on a flat plane using a `<canvas>` (maybe using ThreeJS) instead. 
3. Apply the Burn-My-Windows shader to the plane.

A quick Google search brought up this one: https://codepen.io/ksenia-k/full/yLQLyZa I think this does most of the required things.

Maybe you could start with a simple effect, such as the TV effect (https://github.com/Schneegans/Burn-My-Windows/blob/main/resources/shaders/tv.frag). If this works, all the other effects should work as well. Many require some additional uniform values for tweaking the effect and some require additional texture input but this shouldn't be a problem.

The only issue is that Burn-My-Windows has to be released under the GPL3 license as it directly uses GNOME Shell code. However, if you manage to make this work, I would dual-license the shader code under the MIT license. You will only need the shader code, the other source files of the extension will not be of much help for you, I guess :wink:

**Reply by DustinBrett** [reply]
Thanks very much! I appreciate the info and the links! That Codepen looks very promising. I will see what I can do!

**Comment by DustinBrett** [comment]
@Schneegans Took a while to get to this but I finally ported these effects over (with the help of AI). ([e0dbacb](https://github.com/DustinBrett/daedalOS/commit/e0dbacb79df321534451ad6aa406b8e26d557a7a))

Thanks for the help with how to do this.

![chrome_RzhADmJ58e](https://github.com/user-attachments/assets/923317fe-89b4-4e48-99f5-70458c04cb69)

<img width="391" height="565" alt="image" src="https://github.com/user-attachments/assets/ea5e96ee-6b0d-4b35-98b9-e86405d1d683" />

**Comment by Schneegans** [comment]
This is totally awesome! Looks pretty smooth! Did you go for the render-div-to-texture approach? Except for dynamic content like videos it seems to work really well! Have you posted this somewhere already? I think folks will love this addition :smile:

**Reply by DustinBrett** [reply]
Thanks! I haven't posted this yet. It turns the window element into a canvas via [html-to-image](https://github.com/bubkoo/html-to-image) which I already use in daedalOS for various things. Without AI I am not sure I would have got it working, all the GLSL knowledge comes from it. I am currently on the free Max 20x plan for 6 months because of being OSS (https://claude.com/contact-sales/claude-for-oss).

**Reply by DustinBrett** [reply]
You could probably use [closeEffect](https://github.com/DustinBrett/daedalOS/blob/main/utils/closeEffect.ts) & [closeEffectShaders](https://github.com/DustinBrett/daedalOS/blob/main/utils/closeEffectShaders.ts) to make a demo site for the effects. It worked with all `.frag` files, but I only selected the ones I liked the most for my site. I access them directly via cloning this repo and importing them via npm/node_modules.

**Comment by Kran67** [comment]
Hi both,

All effects (+ genie effect) in my ui framework (100% canvas) with Claude code Pro.

Thank Schneegans for your great work.
<img width="800" height="531" alt="effets" src="https://github.com/user-attachments/assets/5253505e-2987-49f4-935a-6e220bcdadc8" />
<img width="800" height="531" alt="effets2" src="https://github.com/user-attachments/assets/cbb59633-fec0-44ad-a215-4e230b633356" />
<img width="800" height="531" alt="effets3" src="https://github.com/user-attachments/assets/0f858451-bccf-4e2a-a08a-de8dd1851ae9" />

<img width="800" height="531" alt="effets4" src="https://github.com/user-attachments/assets/73915795-b471-4909-8b43-ef49e2ba83c9" />
